api/v1alpha4/types.go

```diff
@@ -85,10 +87,12 @@ type Filter struct {
 }
 
 type SubnetParam struct {
-	// The UUID of the network. Required if you omit the port attribute.
+	// Optional UUID of the subnet.
+	// If specified this will not be validated prior to server creation.
+	// If specified, the enclosing `NetworkParam` must also be specified by UUID.
 	UUID string `json:"uuid,omitempty"`
 
-	// Filters for optional network query
+	// Filters for optional subnet query
```

Comment:
So you can search for a subnet by ID (your use case), you just have to do it in the SubnetFilter. Makes sense.